#!/usr/bin/env python
PACKAGE = "bender_arm_control"

from dynamic_reconfigure.parameter_generator_catkin import *
from dynamixel_driver.dynamixel_const import *

gen = ParameterGenerator()

gen.add("speed", double_t, 1, "Joint speed", 1.0, 0.01, 1.0)
gen.add("compliance_margin", int_t, 2, "Compliance margin", 10, DXL_MIN_COMPLIANCE_MARGIN, DXL_MAX_COMPLIANCE_MARGIN)
gen.add("compliance_slope", int_t, 4, "Compliance slope", 250, DXL_MIN_COMPLIANCE_SLOPE, DXL_MAX_COMPLIANCE_SLOPE)
gen.add("punch", int_t, 8, "Punch", 10, DXL_MIN_PUNCH, DXL_MAX_PUNCH)
gen.add("torque_limit", double_t, 16, "Torque limit", 1.0, 0.01, 1.0)
gen.add("torque_enable", bool_t, 32, "Torque enable", True)
gen.add("p_gain", double_t, 64, "P gain", 0.0, 0.0, 100.0)
gen.add("i_gain", double_t, 128, "I gain", 0.0, 0.0, 100.0)
gen.add("d_gain", double_t, 256, "D gain", 0.0, 0.0, 100.0)
gen.add("i_clamp", double_t, 512, "Integral term clamp antiwindup", 0.0, 0.0, 100.0)
gen.add("enable_pid", bool_t, 1024, "PID enable", False)


exit(gen.generate(PACKAGE, "bender_arm_control", "MotorParams"))